pkg/controller/dynamic/dynamic_controller_integration_test.go

```diff
@@ -591,6 +591,11 @@ func testReconcileAcquire(ctx context.Context, t *testing.T, testContext testrun
 		if gcpUnstruct, err = resourceContext.Create(ctx, t, unstructToCreate, systemContext.TFProvider, kubeClient, systemContext.SMLoader, systemContext.DCLConfig, systemContext.DCLConverter); err != nil {
 			t.Fatalf("unexpected error when creating GCP resource '%v': %v", unstructToCreate.GetName(), err)
 		}
+		if unstructToCreate.GroupVersionKind().Kind == "Folder" {
```

Comment:
Can we handle this before the error occurs? i.e. move this block to after initialUnstruct(line 580)

Reply:
Not sure I follow - can you explain why?

Reply:
Ah sorry, I misunderstood the logic. The resource gets created when we call `resourceContext.Create` and we want to hold a bit if it's Folder kind.

I just feel it's a bit hard to read to put this special handling of resource creation inside resourceContext.Get `err != nil` block.
Maybe like:
```
gcpUnstruct, err = resourceContext.Get(ctx, t, unstructToCreate, systemContext.TFProvider, kubeClient, systemContext.SMLoader, systemContext.DCLConfig, systemContext.DCLConverter)
if err != nil {
		if !strings.Contains(err.Error(), "not found") {
			t.Fatalf("unexpected error when GETting '%v': %v", unstructToCreate.GetName(), err)
		}
                if gcpUnstruct, err = resourceContext.Create(ctx, t, unstructToCreate, systemContext.TFProvider, kubeClient, systemContext.SMLoader, systemContext.DCLConfig, systemContext.DCLConverter); err != nil {
			                t.Fatalf("unexpected error when creating GCP resource '%v': %v", unstructToCreate.GetName(), err)
		}
}

// Hold for eventual consistency
if unstructToCreate.GroupVersionKind().Kind == "Folder" {
         // some code...
}
```

Reply:
/lgtm